src/main/java/org/jabref/gui/entryeditor/SourceTab.java

```diff
@@ -201,30 +218,13 @@ public boolean shouldShow(BibEntry entry) {
 
     @Override
     protected void bindToEntry(BibEntry entry) {
-        CodeArea codeArea = createSourceEditor();
-        VirtualizedScrollPane<CodeArea> node = new VirtualizedScrollPane<>(codeArea);
-        NotificationPane notificationPane = new NotificationPane(node);
-        notificationPane.setShowFromTop(false);
-        sourceValidator.getValidationStatus().getMessages().addListener((ListChangeListener<ValidationMessage>) c -> {
-            if (sourceValidator.getValidationStatus().isValid()) {
-                notificationPane.hide();
-            } else {
-                sourceValidator.getValidationStatus().getHighestMessage().ifPresent(validationMessage -> {
-                    notificationPane.show(validationMessage.getMessage());//this seems not working
-                    dialogService.showErrorDialogAndWait(validationMessage.getMessage());
-                });
-            }
-        });
-        this.setContent(codeArea);
-        this.codeArea = codeArea;
-
-        // Store source for on focus out event in the source code (within its text area)
-        // and update source code for every change of entry field values
-        BindingsHelper.bindContentBidirectional(entry.getFieldsObservable(), codeArea.focusedProperty(), onFocus -> {
-            if (!onFocus) {
-                storeSource(entry, codeArea.textProperty().getValue());
-            }
-        }, fields -> {
+        if (boundEntry != null) {
+            storeSource(boundEntry, codeArea.textProperty().getValue());
+        }
+
+        this.boundEntry = entry;
+
+        if (entry != null) {
```

Comment:
Why did you removed the binding `fields -> code area`? If the entry changes (e.g. cleanup actions or "generate bibtex key" or set special fields like ranked/printed/...), then these changes should be reflected in the source tab, too.

Reply:
This was part of the fix to the CodeArea problem, @Siedlerchr mentioned. Selecting another entry copied the content of the sourcetab into the new selected entry, so it duplicated your old one and removed the selected one. But I did not think of the binding to the other fields. 🤦‍♂ 
I'm going to take a look into this tonight.

Reply:
Fixed it.